src/components/common/PriceLabel.tsx

```diff
@@ -60,18 +60,24 @@ const PriceLabel: React.VFC<
 > = ({ variant, render, noFreeText, ...options }) => {
   const { listPrice, salePrice, downPrice, periodAmount, periodType } = options
   const { formatMessage } = useIntl()
-  const { currencyId: appCurrencyId, settings } = useApp()
+  const { currencyId: appCurrencyId, settings, currencies } = useApp()
   const { locale } = useContext(LanguageContext)
 
   const currencyId = options.currencyId || appCurrencyId
+  const minorUnits = currencies[currencyId].minor_units || 0
   const displayPrice = salePrice || listPrice
   const firstPeriodPrice = displayPrice - (downPrice || 0)
 
   const formatPrice = (price: number) => {
     if (currencyId === 'LSC') {
       return price + (settings['coin.unit'] || 'Coins')
     }
-    return price.toLocaleString(locale, { style: 'currency', currency: currencyId, minimumFractionDigits: 0 })
+    return price.toLocaleString(locale, {
+      style: 'currency',
+      currency: currencyId,
+      maximumFractionDigits: minorUnits,
+      minimumFractionDigits: 0,
+    })
```

Comment:
use the `currencyFormatter` in the utils from the other PR
btw, I thought `formatCurrency()` is the better naming